openraft/src/raft_types.rs

```diff
@@ -146,6 +146,7 @@ pub enum Update<T> {
 
 #[derive(Debug, Clone, PartialOrd, PartialEq, Eq, Serialize, Deserialize)]
 pub enum UpdateMetricsOption {
+    NoUpdate,
     Update,
     AsIs,
 }
```

Comment:
As I understand, these three variants represent:

- `NoUpdate`: do not update leader metrics or other metrics

- `Update`: update leader metrics and other metrics

- `AsIs`: do not update leader metrics, only update other metrics

Thus the combinations can be expressed by two standalone fields:

```rust
struct UpdateMetricsOption {
    leader_metrics: Update<()>,
    other_metrics: Update<()>,
}
```

- This would be more compatible with the future metrics structure mentioned in #229
- In `Update<()>` it is possible to use a `()` to indicate that an update should be made, but it does not know what the update is.